File: python/models/point.py
import math
import uuid


class Point:
    def __init__(self, name=None, coordinates=None):
        self.id = self.generate_unique_id()
        self.name = name if name else f"Point_{self.id}"
        self.coordinates = coordinates if coordinates else []
        self.length = self.calculate_length()

    def generate_unique_id(self):
        return str(uuid.uuid4())

    def calculate_length(self):
        return math.sqrt(sum(coord**2 for coord in self.coordinates))

    def __repr__(self):
        return f"Point(id={self.id}, name='{self.name}', coordinates={self.coordinates}, length={self.length})"

File: src/point.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct Point {
    pub id: String,
    pub name: String,
    pub coordinates: Vec<f64>,
    pub labels: HashMap<String, u64>,
}

impl Point {
    pub fn generate_unique_id(&self) -> String {
        uuid::Uuid::new_v4().to_string()
    }

    pub fn calculate_length(point: &Point) -> f64 {
        let sum = point.coordinates.iter().fold(0.0, |acc, x| acc + x * x);
        f64::sqrt(sum)
    }
}
